### ISSUE TITLE
False positive in UnusedMethod (possibly related to magic methods)

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#276]
Expected: no error
Observed: the below errors

```
ERROR: UnusedMethod - src/model.php:12:5 - Method my_class::_modify is never used
    private function _modify(string $name, string $value) : void {


ERROR: PossiblyUnusedMethod - src/model.php:16:5 - Cannot find public calls to method my_class::modify_foo
    public function modify_foo(string $value) : void {
```

Code to reproduce:

```php
<?php
class my_class {
    /** @var string */
    private $value = 'default';
    public function __call(string $name, array $args) {
        if (count($args) == 1) {
            $this->_modify($name, $args[0]);
        }
    }

    // Psalm erroneously warns that this isn't used
    private function _modify(string $name, string $value) : void {
        call_user_func(array($this, 'modify_'.$name), $value);
    }

    public function modify_foo(string $value) : void {
        $this->value = $value;
    }
}
function test_my_class() : void {
    $m = new my_class();
    $m->foo('value');
    $m->modify_foo('value2');
}
test_my_class();
```
